pyro/infer/traceenum_elbo.py

```diff
@@ -133,15 +133,13 @@ def _compute_model_factors(model_trace, guide_trace):
                 cost = site["packed"]["masked_log_prob"]
                 log_factors.setdefault(t, []).append(cost)
                 scales.append(site["scale"])
-    if log_factors:
-        for t, sites_t in enum_sites.items():
-            # TODO refine this coarse dependency ordering using time and tensor shapes.
-            if any(t <= u for u in log_factors):
```

Comment:
The only change is removing line 139. @eb8680 do you have any idea what this line does?

Reply:
Isn't it getting rid of (some , hence the TODO) zero-expectation terms in guide enumeration?

Reply:
Yeah I think it was merely an optimization, but it looks like it can lead to breakage. I'm pretty sure it is correct to add it back. And tests pass 🤷